ui/app/components/modals/cancel-transaction/cancel-transaction.component.js

```diff
@@ -0,0 +1,70 @@
+import React, { PureComponent } from 'react'
+import PropTypes from 'prop-types'
+import Modal from '../../modal'
+import CancelTransactionGasFee from './cancel-transaction-gas-fee'
+import { SUBMITTED_STATUS } from '../../../constants/transactions'
+import { decimalToHex } from '../../../helpers/conversions.util'
+import { getHexGasTotal } from '../../../helpers/confirm-transaction/util'
+
+export default class CancelTransaction extends PureComponent {
+  static contextTypes = {
+    t: PropTypes.func,
+  }
+
+  static propTypes = {
+    createCancelTransaction: PropTypes.func,
+    hideModal: PropTypes.func,
+    showTransactionConfirmedModal: PropTypes.func,
+    transactionStatus: PropTypes.string,
+    defaultNewGasPrice: PropTypes.string,
+  }
+
+  componentDidUpdate () {
+    const { transactionStatus, showTransactionConfirmedModal } = this.props
+
+    if (transactionStatus !== SUBMITTED_STATUS) {
+      showTransactionConfirmedModal()
+      return
+    }
+  }
+
+  handleSubmit = async () => {
+    const { createCancelTransaction, hideModal } = this.props
+
+    await createCancelTransaction()
+    hideModal()
+  }
+
+  handleCancel = () => {
+    this.props.hideModal()
+  }
+
+  render () {
+    const { t } = this.context
+    const { defaultNewGasPrice: gasPrice } = this.props
+    const newGasFee = getHexGasTotal({ gasPrice, gasLimit: decimalToHex(21000) })
```

Comment:
Can we do this calculation in `mapStateToProps` of the container/parent and just pass `newGasFee` as a prop?

Reply:
The reason I did this here was in anticipation of adding the ability for users to customize the gas price increase, so I was thinking that the `gasPrice` will be dynamic and the total gas fee will need to be re-calculated on user input.

Reply:
but can't that recalculation happen in `mapStateToProps` as well?

Reply:
Are you thinking that the container specifies what gasPrice values the user is allowed to pick, and therefore knows what the gasPrice options are in `mapStateToProps`? That makes sense now because the new `gasPrice` is taken from `mapStateToProps` but I'm thinking at some point we'll allow users to specify their own gasPrice. I guess it's unnecessary now so I'll move this into the container and make changes when the time comes.

Reply:
What I was thinking was that even if we allow users to specify their own gas price, the updated price would still be available in `mapStateToProps`.

I guess we would not be able to do that if you were only storing the updated gas price in component state. That would differ from what I am doing with the gas customization modal, where the custom gasPrice and gasLimit are being stored in redux: https://github.com/MetaMask/metamask-extension/blob/new-gas-customize-feature-branch/ui/app/ducks/gas.duck.js

Reply:
Oh yeah, that's the difference. I wasn't considering putting the gasPrice in redux state.